receiver/pulsarreceiver/pulsar_receiver.go

```diff
@@ -302,11 +340,11 @@ func consumeLogsLoop(ctx context.Context, c *pulsarLogsConsumer) error {
 			_ = c.consumer.Ack(message)
 			return err
```

Comment:
you need to end the logsOp here too, and with any early return.

Reply:
Done.